providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,317 @@
+        couldn't find card name
+    OFFLOAD_FAIL : int
+        couldn't find process on specific GPU
+    NOT_SUPPORT_NV_PRIME : int
+        this system couldn't support nv prime offload
+    NV_DRIVER_ERROR : int
+        some errors form nvidia driver
+    """
+
+    NO_ERROR = 0
+    NO_CARD_ID = -1
+    NO_CARD_NAME = -2
+    OFFLOAD_FAIL = -3
+    NOT_SUPPORT_NV_PRIME = -4
+    NV_DRIVER_ERROR = -5
+
+
+class PrimeOffloader:
+    """
+    A class used to execute process to specific GPU.
+    Have to run this as root.
+
+    Attributes
+    ----------
+    logger : obj
+        console logger
+    check_result : int
+        store the result of checking offloading is ok or not.
+
+    Methods
+    -------
+    find_card_id(pci_name)
+        find card id by pci name
+
+    find_card_name(pci_name)
+        find card name by pci name
+
+    check_offload(cmd, card_id, card_name)
+        check offload status
+
+    check_nv_offload_env()
+        check the environment is ok for prime offload
+
+    run_offload_cmd(cmd, pci_name, driver)
+        exectue the command that you would like to offload to specific GPU
+
+    main()
+        main function for command line processing
+    """
+
+    logger = logging.getLogger()
+
+    check_result = PrimeOffloaderError.OFFLOAD_FAIL
+
+    def find_card_id(self, pci_name):
+        """
+        use pci name to find card id under /sys/kernel/debug/dri
+
+        Parameters
+        ----------
+        pci_name: pci device name in NNNN:NN:NN.N format
+        """
+        cmd = "sudo grep -lr --include=name \"{}\""  \
+              " /sys/kernel/debug/dri 2>/dev/null".format(pci_name)
+        try:
+            if not re.match("[0-9]{4}:[0-9]{2}:[0-9]{2}.[0-9]", pci_name):
+                return PrimeOffloaderError.NO_CARD_ID
+
+            card_path = subprocess.run(cmd, shell=True,
+                                       stdout=subprocess.PIPE,
+                                       universal_newlines=True)
+            card_id = card_path.stdout.split('/')[5]
+            return card_id
+        except (ValueError, IndexError):
+            return PrimeOffloaderError.NO_CARD_ID
+
+    def find_card_name(self, pci_name):
+        """
+        use pci name to find card name by lshw
+
+        Parameters
+        ----------
+        pci_name: pci device name in NNNN:NN:NN.N format
+        """
+        cmd = 'sudo lshw -c display -json'
+        try:
+            card_infos = subprocess.run(cmd, shell=True,
+                                        stdout=subprocess.PIPE,
+                                        universal_newlines=True)
+            infos = json.loads(card_infos.stdout)
+            for info in infos:
+                if pci_name in info['businfo']:
+                    return info['product']
+        except (ValueError, IndexError):
+            return None
+
+    def check_offload(self, cmd, card_id, card_name):
+        """
+        checking command is running
```

Comment:
This is a bit too vague of a description. This function checks that a given command is being executed by a specific GPU using the kernel debug interface, right? Could you add some more info for our future selves? :)